stan/math/prim/fun/trace_gen_quad_form.hpp

```diff
@@ -68,10 +59,10 @@ inline return_type_t<TD, TA, TB> trace_gen_quad_form(
  * @throw std::domain_error if A cannot be multiplied by B or B cannot
  * be multiplied by D.
  */
-template <int RD, int CD, int RA, int CA, typename TB, int RB, int CB>
-inline double trace_gen_quad_form(const Eigen::Matrix<double, RD, CD> &D,
-                                  const Eigen::Matrix<double, RA, CA> &A,
-                                  const Eigen::Matrix<double, RB, CB> &B) {
+template <typename TD, typename TA, typename TB,
+          typename = require_all_eigen_t<TD, TA, TB>,
+          typename = require_all_same_vt<double, TD, TA, TB>>
```

Comment:
Should we not do this for all arithmetics?

Reply:
You are right. Done.